src/metatensor/models/utils/data/writers/xyz.py

```diff
@@ -1,33 +1,124 @@
-from typing import List
+from typing import Dict, List
 
 import ase
 import ase.io
+import metatensor.torch
+import numpy as np
 import torch
-from metatensor.torch import TensorMap
-from metatensor.torch.atomistic import System
+from metatensor.torch import Labels, TensorMap
+from metatensor.torch.atomistic import ModelCapabilities, System
 
 
-def write_xyz(filename: str, predictions: TensorMap, systems: List[System]) -> None:
-    """An ase based xyz file writer
+def write_xyz(
+    filename: str,
+    systems: List[System],
+    capabilities: ModelCapabilities,
```

Comment:
This is a bit of a difficult decision. This is the best I could think of... the issue is that, from a dictionary where the names of the predictions are simple strings, we don't have the physical quantities, which are needed to know how to save things properly. However, I think I can extract that logic in some sort of a pre-processor for the outputs that we might be able to use for multiple writers. I'll try

Reply:
EDIT: This doesn't work because it's even messier to rename the gradients and/or detach the gradient block from the base gradient (one would have to re-do all the samples and it would become a maintenance nightmare)

Reply:
Maybe I am wrong, but I think we basically have to possible data to store:

1. the length of the data (regardless if values or gradients) is the same as the number of atoms -> you write this in the `array` attribute of the `ase.Atoms` object.
2. The length is different -> you write this in the `info` section.

This should cover everything and also should be in line with what we expect from forces, energies etc.

Reply:
This is about renaming stuff for the user. For example, changing the name from U0_positions_gradients to `forces[U0]`, changing the sign accordingly, doing the same for stresses and so on. But how do we know that `U0` is an energy? We need its physical quantity

Reply:
And yes, I agree about what goes in arrays and what goes in info